Source/SwiftLintBuiltInRules/Rules/RuleConfigurations/FileNameConfiguration.swift

```diff
@@ -17,3 +17,20 @@ struct FileNameConfiguration: SeverityBasedRuleConfiguration {
     @ConfigurationElement(key: "require_fully_qualified_names")
     private(set) var requireFullyQualifiedNames = false
 }
+
+// MARK: - For `excluded` option
+extension FileNameConfiguration {
+    func shouldExclude(filePath: String) -> Bool {
+        let fileName = filePath.bridge().lastPathComponent
+
+        // For backwards compatibility,
+        // `excluded` can have a fileName which is invalid as regex.(e.g. "NSString+Extension.swift")
+        if excluded.contains(fileName) {
```

Comment:
I see a problem here. Your example `NSString+Extension.swift` wouldn't match against  `NSStringExtension.swift` here but then later in the regex-based match. That's undesired. 😶

Reply:
Good catch.

From input, we cannot distinguish whether `+` means `regex "+"` or `fileName "+"`.
Some possible options I came up with are
* Accept this behavior since it's not a false alerts.
* Add one new configuration param something like `exclude_pattern`, in addition to existing `exclude` param.

The second option seems better since it's working properly (though, the num of param increased) 🤔

Reply:
> * Accept this behavior since it's not a false alerts.

It could cause false negatives though.

> * Add one new configuration param something like `exclude_pattern`, in addition to existing `exclude` param.

I also think that's the better solution. Unfortunately, it leads to yet another option. I was hoping to just extend existing functionality by supporting regex. But that doesn't seem to be possible without changing semantics.

If we add a new option anyway, we can also have it support complete path patterns. Then we not only support regex but also more specific exclusions (for whatever that's good for). With that, the name should include the "path" aspect like `exclude_path_patterns`.